src/infrastructure/network/undiciRequest.ts

```diff
@@ -0,0 +1,26 @@
+import { Client } from 'undici';
```

Comment:
I think we shouldn't call this file "undiciRequest" because the scope is to make the rest of code "library-agnostic" (right?). 
If we change undici with "got" we must rename all import statements everywhere.

Maybe we can simply call it "request.ts".

Reply:
UndiciRequest is our HTTP adapter, it wraps the undici lib inside a function with a library agnostic signature.
`const request: RequestT = async ({ basePath, authorization, method, path, body })`.
If we want to create a new adapter based on got we should to create a new function which implements this signature `const request: RequestT = async ({ basePath, authorization, method, path, body })`. 

Our system is client agnostic, this is ensured by the airtable object signature
`export default ({ apiKey, app, table, request }: AirtableProps): Store => `

Airtable asks to inject a request function, here the airtable factoryfun parameter interface

`interface AirtableProps {
  apiKey: string;
  app: string;
  table: string;
  request: RequestT;
}`

As you can see we inject a function which implement RequestT, it can be an undiciRequest, a gotRequest or an axiosRequest